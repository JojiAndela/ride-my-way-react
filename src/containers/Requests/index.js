import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getRequest, decideRequest } from 'Store/actions/rides';
import classes from './styles.scss';
/**
 * Requests
 */
class Requests extends Component {
  componentDidMount(){
    this.props.getRequestAction()
  }

  checkBtn = (status, ride, request) => {
    if (status === "pending request") {
      return(
        <div>
        <a onClick={() => this.props.decideRequestAction(ride, request, true) } className={classes.btn_green}> Accept</a>
      <a onClick={() => this.props.decideRequestAction(ride, request, false)} className={classes.btn_red}> Reject</a>
    </div>
        );
      } else {
        return <p>{status}</p>;
      }
  }
  render() {
    const { requestSent, requestRecieve } = this.props.requests;

    let ReqReceived = <p>No Request</p>
    let ReqSent = <p>No Request</p>
    if(requestRecieve && requestRecieve.length){
      ReqReceived = requestRecieve.map(req =>{
        return <div key={req.id} className={classes.single}>
          <div>
        <h3>{req.ride_name}</h3>
        <p>{req.rider}</p>
    </div>
    <div className={classes.btn}>
      {this.checkBtn(req.status, req.ride_id, req.id)}
    </div>
        </div>
        });
    }
    if(requestSent && requestSent.length){
      ReqSent = requestSent.map(req =>{
        return <div key={req.id} className={classes.single}>
          <h3>{req.ride_name}</h3>
        <p>{req.status}</p>
        </div>
        });
    }
    console.log(this.props.requests);
    return (
      <div className={classes.Requests}>
        <div className={classes.card}>
          <h3> Ride Request Sent To You </h3>
        <hr />
      {ReqReceived}
        </div>
        <div className={classes.card}>
          <h3>Request Request You sent</h3>
          <hr />
        {ReqSent}
        </div>
      </div>
    );
  }
}


Requests.propTypes = {
  requests: PropTypes.object.isRequired,
  getRequestAction: PropTypes.func.isRequired
}
const mapStateToProps = state => ({
  requests: state.rides.requests,
});

const mapDispatchToProps = dispatch => ({
  getRequestAction: () => dispatch(getRequest()),
  decideRequestAction: (ride, request, state) => dispatch(decideRequest(ride, request, state)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Requests);
