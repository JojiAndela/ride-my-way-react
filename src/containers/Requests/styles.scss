.Requests{
  .card{
    width: 90%;
    margin: 20px auto;
    padding: 20px;
    border: 1px #ccc solid;
    border-radius: 5px;

    .single{
      padding: 10px;
      display: inline-grid;
      grid-template-columns: 70% 30%;
      width: 97%;
      &:hover{
        background: #faf5f5;
      }
    }
  }
}

.btn_red{
  cursor: pointer;
  padding: 10px;
  color: #fff;
  background-color: red;
  margin: 10px;
  border-radius: 5px;
  font-weight: bold;
  &:hover{
    background-color: #a20505;
  }
}

.btn_green{
  cursor: pointer;
  padding: 10px;
  color: #fff;
  background-color: green;
  margin: 10px;
  border-radius: 5px;
  font-weight: bold;
  &:hover{
    background-color: #194f04;
  }
}
.btn{
  padding: 35px 0;
  text-align: right;
  display: flex;
  text-align: center;
}

@media screen and (max-width: 700px){
  .Requests{
    width: 90%;
    padding: 2.5%;
  }
  .btn{
    flex-direction: column;
    padding: 0;
    font-size: 70%;
  }
}
