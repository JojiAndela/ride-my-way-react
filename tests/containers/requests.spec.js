import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

// Component to be tested
import Requests from '../../src/containers/Requests';

const initialState ={
  rides: {
    loading: false,
    rides: [],
    singleRide: null,
    requests: {
      requestSent:[{
        id: 1,
        ride_name: 'name',
        rider:'rider',
        status: 'status',
        ride_id: '2'
      }],
      requestRecieve: [{
        id: 1,
        ride_name: 'name',
        rider:'rider',
        status: 'status',
        ride_id: '2'
      }]
    },
    myRequest: {}
  }
}


const middleware = [thunk];
const mockStore = configureStore(middleware);
let wrapper;
let store;
describe('Requests', () => {
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <BrowserRouter>
          <Requests
          />
        </BrowserRouter>
      </Provider>
    );
  });
  it('should render the rides page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
