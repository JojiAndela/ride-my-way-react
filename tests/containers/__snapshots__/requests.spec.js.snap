// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requests should render the rides page correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Requests)>
        <Requests
          decideRequestAction={[Function]}
          getRequestAction={[Function]}
          requests={
            Object {
              "requestRecieve": Array [
                Object {
                  "id": 1,
                  "ride_id": "2",
                  "ride_name": "name",
                  "rider": "rider",
                  "status": "status",
                },
              ],
              "requestSent": Array [
                Object {
                  "id": 1,
                  "ride_id": "2",
                  "ride_name": "name",
                  "rider": "rider",
                  "status": "status",
                },
              ],
            }
          }
        >
          <div
            className="Requests"
          >
            <div
              className="card"
            >
              <h3>
                 Ride Request Sent To You 
              </h3>
              <hr />
              <div
                className="single"
                key="1"
              >
                <div>
                  <h3>
                    name
                  </h3>
                  <p>
                    rider
                  </p>
                </div>
                <div
                  className="btn"
                >
                  <p>
                    status
                  </p>
                </div>
              </div>
            </div>
            <div
              className="card"
            >
              <h3>
                Request Request You sent
              </h3>
              <hr />
              <div
                className="single"
                key="1"
              >
                <h3>
                  name
                </h3>
                <p>
                  status
                </p>
              </div>
            </div>
          </div>
        </Requests>
      </Connect(Requests)>
    </Router>
  </BrowserRouter>
</Provider>
`;
